### PR TITLE
Integrated Resume parser code into main website

### PR DESCRIPTION
**Pull request by stuti2731**
**1. Resume Parser Implementation in the main website:**

-     Added a new feature for parsing resumes uploaded by users.

-     Supports both .docx and .pdf formats.

-     Extracts important keywords (e.g., programming languages, skills, etc.) from the resume text.

-     Displays the keywords found in the uploaded resume to the user, or a message indicating no keywords found.

- **Need to add more keywords, more testing is required.**

**2. Styling Enhancements:**

-     Applied CSS styling to the resume parser page for a cleaner user interface.

-     Updated the sidebar to include Resume Parser as a tab.

-    Created a custom stylesheet (style_resume.css) for the resume parser page to ensure consistency in design with the rest of the application.

- **On accessing search tab, error is shown. Need to fix that.**

This is what Resume parser tab looks like:
image

### CODE REVIEW

File: src/backend/__pycache__/rag.cpython-310.pyc
Comment (disha101003) [inline src/backend/__pycache__/rag.cpython-310.pyc:1]: Can you  remove cache files marked with __pycache__ ? We do not want  that on the github repository ?

File: src/backend/app/__pycache__/__init__.cpython-310.pyc
Comment (disha101003) [inline src/backend/app/__pycache__/__init__.cpython-310.pyc:1]: Can you remove cache files ? just push after deleting and commiting.

File: src/backend/app/__pycache__/app.cpython-310.pyc
Comment (disha101003) [inline src/backend/app/__pycache__/app.cpython-310.pyc:1]: Similiarly remove this cache file as well.

File: src/backend/app/__pycache__/__init__.cpython-310.pyc
Comment (stuti2731) [reply src/backend/app/__pycache__/__init__.cpython-310.pyc:1]: Done

File: src/backend/app/__pycache__/app.cpython-310.pyc
Comment (stuti2731) [reply src/backend/app/__pycache__/app.cpython-310.pyc:1]: Done

File: src/backend/__pycache__/rag.cpython-310.pyc
Comment (stuti2731) [reply src/backend/__pycache__/rag.cpython-310.pyc:1]: Done

File: extract_license.py
Comment (disha101003) [inline extract_license.py:1]: Why do we need this file ? What are we extracting license for ?

Comment (stuti2731) [reply extract_license.py:1]: It's for SOBM deliverable.

Comment (disha101003) [reply extract_license.py:1]: ok got it, thanks

File: src/backend/app/app.py [hunk 1]
```diff
@@ -25,6 +38,48 @@ def create_table():
...
+    """Check if the uploaded file is allowed."""
+    return '.' in filename and filename.rsplit('.', 1)[1].lower() in app.config['ALLOWED_EXTENSIONS']
+
+def extract_text_from_docx(file_path):
+    """Extract text from a .docx file."""
+    doc = docx.Document(file_path)
+    return "\n".join(paragraph.text for paragraph in doc.paragraphs)
+
+
+def extract_keywords(text):
```
Comment (disha101003) [inline src/backend/app/app.py:54]: The list of potential keywords looks effective and diverse, good work

File: src/frontend/static/css/style_resume.css
Comment (disha101003) [inline src/frontend/static/css/style_resume.css:1]: looks good to me.

File: src/backend/app/app.py
Comment (disha101003) [inline src/backend/app/app.py:1]: Did you try checking for PEP8 formatting (testing with flake) after you made the changes ?

File: src/frontend/templates/resume_parser.html
Comment (disha101003) [inline src/frontend/templates/resume_parser.html:1]: looks perfect to me, whenever you get time just add comments as well so that the code is documented.

**Comment by disha101003** [comment]
Can you add the dependencies for resume parser in requirements.txt ?

File: src/backend/app/app.py [hunk 2]
```diff
@@ -57,13 +112,13 @@ def login():
         email = request.form['email']
         password = request.form['password']
 
-        # Retrieve the user from the database
```
Comment (disha101003) [inline src/backend/app/app.py:60]: why did u delete my comments ?

File: src/backend/app/app.py [hunk 3]
```diff
@@ -57,13 +112,13 @@ def login():
...
+       
         with get_db() as conn:
             user = conn.execute('SELECT * FROM users WHERE email = ?', (email,)).fetchone()
 
         if user and user['password'] == password:
             flash(f"Welcome back, {user['username']}!", 'success')
-            return redirect(url_for('queryforge'))  # Redirect to the front page
+            return redirect(url_for('queryforge'))  
         else:
             flash('Invalid credentials! Please try again.', 'danger')
```
Comment (disha101003) [inline src/backend/app/app.py:123]: Found the problem, index and login routes got merged here. Fixed it

File: src/backend/app/app.py [hunk 4]
```diff
@@ -1,16 +1,29 @@
 from flask import Flask, render_template, request, redirect, url_for, flash
 import sqlite3
```
Comment (disha101003) [inline src/backend/app/app.py:2]: I think you removed all the existing imports while merging , cause I cant find RAG imports  and changed the database as well . I am rewriting this file.

File: src/frontend/static/css/style_resume.css [hunk 5]
```diff
@@ -0,0 +1,124 @@
+/* Style for the entire page */
```
Comment (disha101003) [inline src/frontend/static/css/style_resume.css:1]: I will move this file along with all the other css files

File: src/backend/app/app.py (hunk 1 above)
Comment (stuti2731) [reply src/backend/app/app.py:54]: Thanks!

File: src/frontend/static/css/style_resume.css
Comment (stuti2731) [reply src/frontend/static/css/style_resume.css:1]: Ok

File: src/frontend/static/css/style_resume.css (hunk 5 above)
Comment (stuti2731) [reply src/frontend/static/css/style_resume.css:1]: Okay

**Comment by stuti2731** [comment]
Screenshot from 2024-12-03 14-24-45
Screenshot from 2024-12-03 14-25-01
This is a sample working of resume parser.

File: src/backend/app/app.py (hunk 3 above)
Comment (stuti2731) [reply src/backend/app/app.py:123]: Okay.

File: src/backend/app/app.py (hunk 4 above)
Comment (stuti2731) [reply src/backend/app/app.py:2]: Okay, thanks!

File: src/backend/app/app.py (hunk 2 above)
Comment (stuti2731) [reply src/backend/app/app.py:60]: I might have done that by mistake. Can you add again?

**Pull request merged**